File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Shapes/extractimages.rb
module Asposediagramjava
  module ExtractImages
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        shapes = diagram.getPages().getPage("Flow 1").getShapes()
        
        i = 0
        while i < shapes.getCount()
            shape = shapes.get(i)
            # Filter shapes by type Foreign
            if shape.getType() == Rjb::import('com.aspose.diagram.TypeValue').FOREIGN
                # create an image file
                fos = Rjb::import('java.io.FileOutputStream').new(data_dir + "Image#{i}.bmp")
                fos.write(shape.getForeignData().getValue())
                fos.close()
            end

            i +=1
        end

        puts "Extracted images successfully!"
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithShapes/ExtractImages.php
<?php
namespace Aspose\Diagram\WorkingwithShapes;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
use com\aspose\diagram\TypeValue as TypeValue;

use java\io\FileOutputStream as FileOutputStream;
class ExtractImages{

    public static function run($dataDir=null)
    {

        # Create instance of Diagram
        $diagram = new Diagram($dataDir."drawing.vsd");

        $shapes = $diagram->getPages()->getPage("Flow 1")->getShapes();
        $i = 0;
        $typeValue=new TypeValue();
        while ($i <(int)(string)$shapes->getCount()){
        $shape = $shapes->get($i);
        # Filter shapes by type Foreign
        if ($shape->getType()== $typeValue->FOREIGN){
        # create an image file
        $fos = new FileOutputStream($dataDir."Image#{i}.bmp");
        $fos->write($shape->getForeignData()->getValue());
        $fos->close();
        }
        $i += 1;
        }
        print "Extracted images successfully!".PHP_EOL;
    }
}